homeassistant/components/airzone/binary_sensor.py

```diff
@@ -0,0 +1,112 @@
+"""Support for the Airzone sensors."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from dataclasses import dataclass
+from typing import Any, Final
+
+from aioairzone.const import (
+    AZD_AIR_DEMAND,
+    AZD_DEMAND,
+    AZD_ERRORS,
+    AZD_FLOOR_DEMAND,
+    AZD_NAME,
+    AZD_PROBLEMS,
+    AZD_ZONES,
+)
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_PROBLEM,
+    DEVICE_CLASS_RUNNING,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import AirzoneEntity
+from .const import DOMAIN
+from .coordinator import AirzoneUpdateCoordinator
+
+
+@dataclass
+class AirzoneBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes airzone binary sensor entities."""
+
+    attributes: dict[str, str] | None = None
+
+
+BINARY_SENSOR_TYPES: Final[tuple[AirzoneBinarySensorEntityDescription, ...]] = (
+    AirzoneBinarySensorEntityDescription(
+        attributes={
+            "air": AZD_AIR_DEMAND,
+            "floor": AZD_FLOOR_DEMAND,
+        },
+        device_class=DEVICE_CLASS_RUNNING,
+        key=AZD_DEMAND,
+        name="Demand",
+    ),
+    AirzoneBinarySensorEntityDescription(
+        attributes={
+            "errors": AZD_ERRORS,
+        },
```

Comment:
What do these attributes represent?

Reply:
The array of errors returned by the API for each HVAC Zone.
If there's at least one entry in the array, then the binary sensor will be enabled, but this way the user can check what the specific error is.

Reply:
If its a smaller fixed of potentially problems we want a problem binary sensor per error

If its a large list or non-binary, it should be its own sensor entity.

Reply:
It's a very large list:
![image](https://user-images.githubusercontent.com/264346/160290206-0d79cdd7-e5a7-49be-8078-2499e238bd7b.png)
![image](https://user-images.githubusercontent.com/264346/160290216-4653f106-2fdf-448b-9f66-b2b5b2c67d9c.png)

Reply:
That's a very strange implementation of error reporting.  I think its best to leave this as-is.